modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/DLMimeTypeDisplayContextUtil.java

```diff
@@ -15,43 +15,28 @@
 package com.liferay.wiki.web.internal.display.context;
 
 import com.liferay.document.library.display.context.DLMimeTypeDisplayContext;
+import com.liferay.osgi.util.service.Snapshot;
 import com.liferay.portal.kernel.repository.model.FileVersion;
 
-import org.osgi.framework.BundleContext;
-import org.osgi.service.component.annotations.Activate;
-import org.osgi.service.component.annotations.Component;
-import org.osgi.service.component.annotations.Deactivate;
-import org.osgi.service.component.annotations.Reference;
-
 /**
  * @author Eudaldo Alonso
  */
-@Component(service = {})
 public class DLMimeTypeDisplayContextUtil {
 
 	public static String getCssClassFileMimeType(FileVersion fileVersion) {
-		return _dlMimeTypeDisplayContextUtil._dlMimeTypeDisplayContext.
-			getCssClassFileMimeType(fileVersion.getMimeType());
+		return _dlMimeTypeDisplayContext.getCssClassFileMimeType(
+			fileVersion.getMimeType());
 	}
 
 	public static String getIconFileMimeType(FileVersion fileVersion) {
-		return _dlMimeTypeDisplayContextUtil._dlMimeTypeDisplayContext.
-			getIconFileMimeType(fileVersion.getMimeType());
-	}
-
-	@Activate
-	protected void activate(BundleContext bundleContext) {
-		_dlMimeTypeDisplayContextUtil = this;
+		return _dlMimeTypeDisplayContext.getIconFileMimeType(
+			fileVersion.getMimeType());
 	}
 
-	@Deactivate
-	protected void deactivate() {
-		_dlMimeTypeDisplayContextUtil = null;
-	}
-
-	private static DLMimeTypeDisplayContextUtil _dlMimeTypeDisplayContextUtil;
-
-	@Reference
-	private DLMimeTypeDisplayContext _dlMimeTypeDisplayContext;
+	private static final DLMimeTypeDisplayContext _dlMimeTypeDisplayContext =
+		DLMimeTypeDisplayContextUtil._dlMimeTypeDisplayContextSnapshot.get();
```

Comment:
Hey @RenanUmV, you do not need this field `_dlMimeTypeDisplayContext`. You just need to use the snapshot in the methods implementations.

Thx